Minio/DataModel/ObjectArgs.cs

```diff
@@ -23,16 +25,23 @@ public abstract class ObjectArgs<T> : BucketArgs<T>
 
     internal string ObjectName { get; set; }
     internal byte[] RequestBody { get; set; }
+    internal int ExactBodySize { get; set; }
 
     public T WithObject(string obj)
     {
         ObjectName = obj;
         return (T)this;
     }
 
-    public T WithRequestBody(byte[] data)
+    public T WithRequestBody(byte[] data, int exactBodySize = 0)
     {
-        RequestBody = data;
+        if (data is null) return (T)this;
+
+        ExactBodySize = exactBodySize;
+        RequestBody = new byte[exactBodySize];
```

Comment:
Since this is a public API, be sure to check the parameter for valid values. If the value is less than zero, then `throw new ArgumentOutOfRangeException(nameof(exactBodySize), "Exact body size must be greater or equal to zero.")`

Reply:
@pbolduc 

Ok. Will do. 

...and thank you for all your review commands.
btw, there are more changes coming up I introduced while I was testing the memory allocation during uploads for different sizes of objects.